Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,1875 @@
+        return CompletableFuture.supplyAsync(() -> {
+            if (hasAccount(identifier)) return false;
+            return Economy.createNPC(getName(identifier));
+        });
+    }
+
+    /**
+     * Attempts to delete an account for this identifier. This method should be used for non-player accounts.
+     * @param identifier The identifier of the account.
+     * @return True if an account was deleted, else false.
+     */
+    @Override
+    public boolean deleteAccount(String identifier) {
+        try {
+            Economy.resetBalance(identifier);
+        } catch(UserDoesNotExistException | NoLoanPermittedException ignore) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Attempts to delete an account for this identifier. This method should be used for player accounts.
+     * @param identifier The {@link UUID} of the account.
+     * @return True if an account was deleted, else false.
+     */
+    @Override
+    public boolean deleteAccount(UUID identifier) {
+        try {
+            Economy.resetBalance(getName(identifier));
+        } catch(UserDoesNotExistException | NoLoanPermittedException ignore) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Attempts to delete an account for this identifier. This method should be used for non-player accounts.
+     * @param identifier The identifier of the account.
+     * @return True if an account was deleted, else false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncDeleteAccount(String identifier) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Economy.resetBalance(identifier);
+            } catch(UserDoesNotExistException | NoLoanPermittedException ignore) {
+                return false;
+            }
+            return true;
+        });
+    }
+
+    /**
+     * Attempts to delete an account for this identifier. This method should be used for player accounts.
+     * @param identifier The {@link UUID} of the account.
+     * @return True if an account was deleted, else false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncDeleteAccount(UUID identifier) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Economy.resetBalance(getName(identifier));
+            } catch(UserDoesNotExistException | NoLoanPermittedException ignore) {
+                return false;
+            }
+            return true;
+        });
+    }
+
+    /**
+     * Determines whether or not a player is able to access this account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param accessor The identifier of the user attempting to access this account.
+     * @return Whether or not the player is able to access this account.
+     */
+    @Override
+    public boolean isAccessor(String identifier, String accessor) {
+        return false;
```

Comment:
Ehrm shouldn't a player be able to access their own account? If this isn't called in this instance, can you note this in the javadoc?